code/modules/mob/living/carbon/damage_procs.dm

```diff
@@ -0,0 +1,191 @@
+		if(TOX)
+			adjustToxLoss(damage * hit_percent)
+		if(OXY)
+			adjustOxyLoss(damage * hit_percent)
+		if(CLONE)
+			adjustCloneLoss(damage * hit_percent)
+		if(STAMINA)
+			adjustStaminaLoss(damage * hit_percent)
+	return 1
+
+
+//These procs fetch a cumulative total damage from all bodyparts
+/mob/living/carbon/getBruteLoss()
+	var/amount = 0
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		amount += BP.brute_dam
+	return amount
+
+/mob/living/carbon/getFireLoss()
+	var/amount = 0
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		amount += BP.burn_dam
+	return amount
+
+
+/mob/living/carbon/adjustBruteLoss(amount, updating_health = 1)
+	if(status_flags & GODMODE)
+		return 0
+	if(amount > 0)
+		take_overall_damage(amount, 0)
+	else
+		heal_overall_damage(-amount, 0, 0, 1, updating_health)
+
+/mob/living/carbon/adjustFireLoss(amount)
+	if(status_flags & GODMODE)
+		return 0
+	if(amount > 0)
+		take_overall_damage(0, amount)
+	else
+		heal_overall_damage(0, -amount)
+
+
+/mob/living/carbon/adjustToxLoss(amount, updating_health=1)
+	if(has_dna() && TOXINLOVER in dna.species.specflags) //damage becomes healing and healing becomes damage
+		amount = -amount
+		if(amount > 0)
+			blood_volume -= 5*amount
+		else
+			blood_volume -= amount
+	return ..()
+
+////////////////////////////////////////////
+
+//Returns a list of damaged bodyparts
+/mob/living/carbon/proc/get_damaged_bodyparts(brute, burn)
+	var/list/obj/item/bodypart/parts = list()
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if((brute && BP.brute_dam) || (burn && BP.burn_dam))
+			parts += BP
+	return parts
+
+//Returns a list of damageable bodyparts
+/mob/living/carbon/proc/get_damageable_bodyparts()
+	var/list/obj/item/bodypart/parts = list()
+	for(var/X in bodyparts)
+		var/obj/item/bodypart/BP = X
+		if(BP.brute_dam + BP.burn_dam < BP.max_damage)
+			parts += BP
+	return parts
+
+//Heals ONE bodypart randomly selected from damaged ones.
+//It automatically updates damage overlays if necessary
+//It automatically updates health status
+/mob/living/carbon/heal_bodypart_damage(brute, burn, only_robotic = 0, only_organic = 1)
+	var/list/obj/item/bodypart/parts = get_damaged_bodyparts(brute,burn)
+	if(!parts.len)
+		return
+	var/obj/item/bodypart/picked = pick(parts)
+	if(picked.heal_damage(brute, burn, only_robotic, only_organic))
+		update_damage_overlays()
+
+//Damages ONE bodypart randomly selected from damagable ones.
+//It automatically updates damage overlays if necessary
+//It automatically updates health status
+/mob/living/carbon/take_bodypart_damage(brute, burn)
+	var/list/obj/item/bodypart/parts = get_damageable_bodyparts()
+	if(!parts.len)
+		return
+	var/obj/item/bodypart/picked = pick(parts)
+	if(picked.take_damage(brute,burn))
+		update_damage_overlays()
+
+
+//Heal MANY bodyparts, in random order
+/mob/living/carbon/heal_overall_damage(brute, burn, only_robotic = 0, only_organic = 1, updating_health = 1)
```

Comment:
On the testmerged sybil server, this broke. Lightgeists use heal_overall_damage to heal people, and any humans are completely unable to be healed by them.

It's also apparently one of the very few uses of heal_overall_damage, but hey!